### ISSUE TITLE
document and revisit `taskType`

### ISSUE THREAD

**Issue by jkowalleck** [CycloneDX/specification#245]
_Originally posted by @jkowalleck in https://github.com/CycloneDX/specification/pull/222#discussion_r1211618704_
            

see also: https://github.com/CycloneDX/specification/pull/222#issuecomment-1570077893

@mrutkows reminded: should reflect CI/CD stages. There might be some standardization work done for the wordings.

this ticket is about adding documentation for TaskType and its Enum values.
see https://github.com/CycloneDX/specification/blob/8af880d5f2ba0a107de88a920a76cedd5ba75083/schema/bom-1.5.schema.json#L3753

**Comment by stevespringett** [comment]
Is work being done on this for v1.6?

**Comment by jkowalleck** [comment]
@mrutkows  could you provide the missing texts/docs for `taskType`? 
IIRC: You said you would provide them after 1.5 release, because you missed to do so before and ran out of time back then. 

@stevespringett i created the ticket as a reminder for CWG, that the docs were missing in 1.5

**Comment by mrutkows** [comment]
See example: https://github.com/CycloneDX/specification/blob/46ff114b2bf497da0286b4ce31082cc2739b5cf5/schema/bom-1.6.schema.json#L605

```
                    "enum": [
                      "design",
                      "pre-build",
                      "build",
                      "post-build",
                      "operations",
                      "discovery",
                      "decommission"
                    ],
                    "meta:enum": {
                      "design": "BOM produced early in the development lifecycle containing inventory of components and services that are proposed or planned to be used. The inventory may need to be procured, retrieved, or resourced prior to use.",
                      "pre-build": "BOM consisting of information obtained prior to a build process and may contain source files and development artifacts and manifests. The inventory may need to be resolved and retrieved prior to use.",
                      "build": "BOM consisting of information obtained during a build process where component inventory is available for use. The precise versions of resolved components are usually available at this time as well as the provenance of where the components were retrieved from.",
                      "post-build": "BOM consisting of information obtained after a build process has completed and the resulting components(s) are available for further analysis. Built components may exist as the result of a CI/CD process, may have been installed or deployed to a system or device, and may need to be retrieved or extracted from the system or device.",
                      "operations": "BOM produced that represents inventory that is running and operational. This may include staging or production environments and will generally encompass multiple SBOMs describing the applications and operating system, along with HBOMs describing the hardware that makes up the system. Operations Bill of Materials (OBOM) can provide full-stack inventory of runtime environments, configurations, and additional dependencies.",
                      "discovery": "BOM consisting of information observed through network discovery providing point-in-time enumeration of embedded, on-premise, and cloud-native services such as server applications, connected devices, microservices, and serverless functions.",
                      "decommission": "BOM containing inventory that will be, or has been retired from operations."
                    }
```

**Comment by jkowalleck** [comment]
re https://github.com/CycloneDX/specification/issues/245#issuecomment-1932255820
@mrutkows could you fork this repo and do the changes in a feature branch from `1.6-dev`, and open a pull request targeting that same branch.

**Comment by mrutkows** [comment]
@jkowalleck I had to delete my stale fork and create a new one yesterday ;)

**Comment by jkowalleck** [comment]
@mrutkows this might help in the future: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

**Comment by mrutkows** [comment]
> re [#245 (comment)](https://github.com/CycloneDX/specification/issues/245#issuecomment-1934671821) @mrutkows this might help in the future: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

Unfortunately, the new GitHub GUI only allows syncing for existing forks...  and the command line would have me compose commands on a per-branch basis...

**Comment by mrutkows** [comment]
Here is what I intend to add if I can figure out how to properly commit to the branch...

```
      "meta:enum": {
        "copy": "A task that copies software or data used to accomplish other tasks in the workflow.",
        "clone": "A task that clones a software repository into the workflow in order to retrieve its source code or data for use in a build step.",
        "lint": "A task that checks source code for programmatic and stylistic errors.",
        "scan": "A task that performs a scan against source code, or built or deployed components and services. Scans are typically run to gather or test for security vulnerabilities or policy compliance.",
        "merge": "A task that merges changes or fixes into source code prior to a build step in the workflow.",
        "build": "A task that builds the source code, dependencies and/or data into an artifact that can be deployed to and executed on target systems.",
        "test": "A task that verifies the functionality of a component or service.",
        "deliver": "A task that delivers a built artifact to one or more target repositories or storage systems.",
        "deploy": "A task that deploys a built artifact for execution on one or more target systems.",
        "release": "A task that releases a built, versioned artifact to a target repository or distribution system.",
        "clean": "A task that cleans unnecessary tools, build artifacts and/or data from workflow storage.",
        "other": "A workflow task that does not match current task type definitions."
      }

```

**Comment by jkowalleck** [comment]
closed via #377
